google/cloud/internal/rest_client.h

```diff
@@ -42,26 +43,50 @@ std::unique_ptr<RestClient> MakeDefaultRestClient(std::string endpoint_address,
 std::unique_ptr<RestClient> MakePooledRestClient(std::string endpoint_address,
                                                  Options options);
 
-// Provides methods corresponding to HTTP verbs to make requests to RESTful
-// services.
+/**
+ * Provides methods corresponding to HTTP verbs to make HTTP requests.
+ *
+ * Concrete versions of this class make HTTP requests. While typically used with
+ * RESTful services, the interface and implementions can be used to make any
+ * HTTP requests.
+ *
+ * The headers, payload, and query parameters for the request are passed in as
+ * a `RestRequest` parameter.  The result is a
+ * `StatusOr<std::unique_ptr<RestResponse>>`. On success, the `RestResponse`
+ * contains the HTTP status code, response headers, and an object to iterate
+ * over the payload.
+ *
+ * Note that HTTP requests that fail with a HTTP status code, e.g. with
```

Comment:
Nit: Typically, I believe we would say "an HTTP foo" rather than "a HTTP foo".

Reply:
The both sound Okay-ish to my ear. Changed it in the places I could find.